packages/yoastseo/src/scoring/assessments/readability/TextAlignmentAssessment.js

```diff
@@ -0,0 +1,153 @@
+		return longCenterAlignedTexts.map( longCenterAlignedText => {
+			const text = longCenterAlignedText.text;
+			const fieldsToMark = longCenterAlignedText.typeOfBlock;
+			/*
+			 * Strip the HTML tags before applying the yoastmark.
+			 * This is because applying yoastmark tags to unsanitized text could lead to highlighting problem(s).
+			 */
+			const marked = addMark( stripHTMLTags( text ) );
+
+			return new Mark( {
+				original: text,
+				marked: marked,
+				fieldsToMark: fieldsToMark,
+			} );
+		} );
+	}
+
+	/**
+	 * Checks whether the assessment is applicable.
+	 * The assessment is applicable when the paper has at least 50 characters (after sanitation)
+	 * and when the researcher has `getLongCenterAlignedText` research.
+	 *
+	 * @param {Paper}       paper       The paper to use for the assessment.
+	 * @param {Researcher}  researcher   The researcher used in the assessment.
+	 *
+	 * @returns {boolean} True when there is text.
+	 */
+	isApplicable( paper, researcher ) {
+		return this.hasEnoughContentForAssessment( paper ) && researcher.hasResearch( "getLongCenterAlignedText" );
+	}
+
+	/**
+	 * Calculates the result based on the number of center-aligned text found in the paper.
+	 *
+	 * @param {Paper}       paper                   The Paper object to assess.
+	 * @param {Researcher}  researcher              The researcher used in the assessment.
+	 * @param {array} textContainsCenterAlignedText The array containing each paragraph or heading
+	 * with center aligned text that’s longer than 50 characters.
+	 *
+	 * @returns {Object} The calculated result.
+	 */
+	calculateResult( paper, researcher, textContainsCenterAlignedText ) {
+		const preferredAlignment = paper.isRTL() ? __( "right-aligned", "wordpress-seo" ) : __( "left-aligned", "wordpress-seo" );
+
+		if ( textContainsCenterAlignedText.length === 1 ) {
+			return {
+				score: this._config.scores.bad,
+				resultText: sprintf(
+					/* Translators: %1$s and %2$s expand to links on yoast.com, %3$s expands to the anchor end tag,
+					%4$s expands to "right-aligned" when the string is shown in an RTL
+					and expands to "left-aligned" when the string is shown in LTR */
+					__(
+						"%1$sAlignment%3$s: Your text has a long block of center-aligned text. %2$sWe recommend changing that to %4$s%3$s.",
+						"wordpress-seo"
+					),
+					this._config.urlTitle,
+					this._config.urlCallToAction,
+					"</a>",
+					preferredAlignment
+				),
+			};
+		}
+
+		return {
+			score: this._config.scores.bad,
+			resultText: sprintf(
+				/* Translators: %1$s and %2$s expand to links on yoast.com, %3$s expands to the anchor end tag,
+				%4$s expands to "right-aligned" when the string is shown in an RTL and expands to "left-aligned" when the string is shown in LTR */
+				__(
+					"%1$sAlignment%3$s: Your text contains multiple long blocks of center-aligned text. %2$sWe recommend changing that to %4$s%3$s.",
```

Comment:
For the feedback for multiple long blocks of center-aligned text, should we show it as shown above, or should we mention the exact number of long blocks of center-aligned text?

So singular:

> Alignment: Your text has 1 long block of center-aligned text. We recommend changing that to left-aligned.

Plural:

> Alignment: Your text has X long blocks of center-aligned text. We recommend changing that to left-aligned.

Let me know what you think :)

Reply:
Adjusted! See the discussion on [Slack](https://yoast.slack.com/archives/C03PRESAXMJ/p1679413754904829?thread_ts=1679406187.807369&cid=C03PRESAXMJ).